python/lib/dependabot/python/file_updater/requirement_replacer.rb

```diff
@@ -137,11 +141,28 @@ def hash_separator(requirement)
         end
 
         def package_hashes_for(name:, version:, algorithm:)
-          SharedHelpers.run_helper_subprocess(
-            command: "pyenv exec python3 #{NativeHelpers.python_helper_path}",
-            function: "get_dependency_hash",
-            args: [name, version, algorithm]
-          ).map { |h| "--hash=#{algorithm}:#{h['hash']}" }
+          index_urls = @index_urls || [nil]
+
+          index_urls.map do |index_url|
+            args = [name, version, algorithm]
+            args << index_url unless index_url.nil?
+
+            begin
+              result = SharedHelpers.run_helper_subprocess(
```

Comment:
Any time I see `SharedHelpers.run_helper_subprocess` being called in a loop, I stop and ask, is there a way we could do this without the loop? The answer might be no, but it's good to keep in mind the amount of overhead this incurs.

Reply:
I thought about that, but since most projects only use PyPI, running the helper twice for the few that use private registries seemed okay to me.